connect/runtime/src/main/java/org/apache/kafka/connect/storage/MemoryOffsetBackingStore.java

```diff
@@ -52,11 +52,15 @@ public void configure(WorkerConfig config) {
 
     @Override
     public synchronized void start() {
+        executor = Executors.newSingleThreadExecutor();
     }
 
     @Override
     public synchronized void stop() {
-        // Nothing to do since this doesn't maintain any outstanding connections/data
+        if (executor != null) {
+            executor.shutdown();
+            executor = null;
```

Comment:
@davispw I think this was just done defensively since the ExecutorService means some state will be mutated in another thread (you'll notice the Kafka*BackingStore implementations aren't synchronized). Since we're only touching the `executor` field here and it should be thread safe anyway, I think we can drop the synchronization.